plugin/core/sessions.py

```diff
@@ -1508,6 +1508,11 @@ def _open_file_uri_async(
         def handle_continuation(view: Optional[sublime.View]) -> None:
             if view and r:
                 center_selection(view, r)
+            elif view:
+                window = view.window()
+                if window:
+                    window.focus_view(view)
```

Comment:
There is one place in LSP code that exercises this branch and that's `apply_workspace_edit_async`. And in that case it makes more sense not to focus all views that are edited IMO.

I feel like there is some refactoring that is needed here to make this all easier to follow and know what to expect. Possibly there should be an explicit argument in all "open file" functions (and there is quite a few) to tell them what to do.

Reply:
I guess one obstacle to making this more predictable is the fact that there is no way to open a new file and **not** give it focus...

Reply:
But that also means that there is currently no way to always focus the view with Session.open_uri_async, which is a quite common use case in my opinion.

Reply:
Maybe apply_workspace_edit_async shouldn't call Session.open_uri_async at all since that is not what I would expect from the method name. I may attempt a refactoring in that regard once I got some spare time.

Reply:
> Maybe apply_workspace_edit_async shouldn't call Session.open_uri_async at all since that is not what I would expect from the method name.

We have to open the file to apply the edit.